lib/deploy/stepFunctions/compileIamRole.js

```diff
@@ -35,12 +35,18 @@ function sqsQueueUrlToArn(serverless, queueUrl) {
     const accountId = match[2];
     const queueName = match[3];
     return `arn:aws:sqs:${region}:${accountId}:${queueName}`;
-  } if (isIntrinsic(queueUrl) && queueUrl.Ref) {
-    // most likely we'll see a { Ref: LogicalId }, which we need to map to
-    // { Fn::GetAtt: [ LogicalId, Arn ] } to get the ARN
-    return {
-      'Fn::GetAtt': [queueUrl.Ref, 'Arn'],
-    };
+  }
+  if (isIntrinsic(queueUrl)) {
+    if (queueUrl.Ref) {
+      // most likely we'll see a { Ref: LogicalId }, which we need to map to
+      // { Fn::GetAtt: [ LogicalId, Arn ] } to get the ARN
+      return {
+        'Fn::GetAtt': [queueUrl.Ref, 'Arn'],
+      };
+    }
+    // in case of for example { Fn::ImportValue: sharedValueToImport }
+    // we need to use "*" as ARN
+    return '*';
```

Comment:
ah, now I see why you want to add both QueueUrl as well as QueueArn, so you can ref the Arn in the IAM instead of QueueUrl. It'd be doable for sure, but it'd be a bit odd though, and hard to explain to people. Let's leave it as it is for now.

Reply:
Yeah, that's what I meant. I agree with you. :slightly_smiling_face: :+1: